snapcraft/internal/build_providers/errors.py

```diff
@@ -0,0 +1,136 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2018 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import shlex
+from typing import Any, Dict
+from typing import Sequence  # noqa: F401
+
+from snapcraft.internal.errors import SnapcraftError as _SnapcraftError
+
+
+class ProviderNotSupported(_SnapcraftError):
+
+    fmt = (
+        'The {provider!r} provider is not supported, please choose a '
+        'different one and try again.'
+    )
+
+    def __init__(self, *, provider: str) -> None:
+        super().__init__(provider=provider)
+
+
+class ProviderCommandNotFound(_SnapcraftError):
+
+    fmt = (
+        'The {command!r} command is necessary to be able to build in this '
```

Comment:
This jumps straight to why this is a problem instead of starting with the actual problem. Consider "{command!r} command not found: this command is necessary to build in this environment.\nInstall {command!r}...".